lib/ts/recipe/passwordless/smsdelivery/services/backwardCompatibility/index.ts

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { TypePasswordlessSmsDeliveryInput } from "../../../types";
+import { SmsDeliveryInterface } from "../../../../../ingredients/smsdelivery/types";
+import { NormalisedAppinfo } from "../../../../../types";
+import axios from "axios";
+import { SUPERTOKENS_SMS_SERVICE_URL } from "../../../../../ingredients/smsdelivery/services/supertokens";
+import Supertokens from "../../../../../supertokens";
+
+function defaultCreateAndSendCustomSms(_: NormalisedAppinfo) {
+    return async (
+        input: {
+            // Where the message should be delivered.
+            phoneNumber: string;
+            // This has to be entered on the starting device  to finish sign in/up
+            userInputCode?: string;
+            // Full url that the end-user can click to finish sign in/up
+            urlWithLinkCode?: string;
+            codeLifetime: number;
+        },
+        _: any
+    ): Promise<void> => {
+        let supertokens = Supertokens.getInstanceOrThrowError();
+        let appName = supertokens.appInfo.appName;
+        try {
+            await axios({
+                method: "post",
+                url: SUPERTOKENS_SMS_SERVICE_URL,
+                data: {
+                    smsInput: {
+                        appName,
+                        type: "PASSWORDLESS_LOGIN",
+                        phoneNumber: input.phoneNumber,
+                        userInputCode: input.userInputCode,
+                        urlWithLinkCode: input.urlWithLinkCode,
+                        codeLifetime: input.codeLifetime,
+                    },
+                },
+                headers: {
+                    "api-version": "0",
+                },
+            });
+            return;
+        } catch (err) {
+            if (err.response === undefined || err.response.status !== 429) {
+                throw err;
```

Comment:
The right message has to be propogated to the user. How is that happening here? I think the message is in err.response.data? So you would need to do `throw new Error(err.response.data)` or something like that?